types/types_test.go

```diff
@@ -84,3 +90,35 @@ func TestTransactionCopy(t *testing.T) {
 		t.Fatal("[ERROR] Copied transaction not equal base transaction")
 	}
 }
+
+func TestAddress_ExistsIn(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name      string
+		address   Address
+		addresses []Address
+		exists    bool
+	}{
+		{
+			name:      "Address exists",
+			address:   addr1,
+			addresses: []Address{addr2, addr1},
+			exists:    true,
+		},
+		{
+			name:      "Address doesn't exists",
+			address:   addr3,
+			addresses: []Address{addr2, addr1},
+			exists:    false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			exists := tc.address.ExistsIn(tc.addresses)
+			assert.Equal(t, exists, tc.exists)
+		})
+	}
+}
```

Comment:
Honestly, I think we can just drop this test entirely. It tests if go knows how to handle for loops 🙂

Reply:
To be honest don't agree, the purpose of this test is to make sure that the feature works, without it i would need to manually test these 3 scenarios to be sure.
If it tested CanDeployContract i would agree, but what if someone delete or change this condition https://github.com/0xPolygon/polygon-edge/blob/feature/permission_smart_contract_deployment/txpool/txpool.go#L552

Reply:
@dbrajovic @Kourin1996 what do you think?

Reply:
What are you talking about - you are literally testing a for loop

Reply:
Agree fixed 88d80daf1ed7db6de97b3a15486a1d9be5cc76f3